src/plots/cartesian/set_convert.js

```diff
@@ -447,6 +447,11 @@ module.exports = function setConvert(ax, fullLayout) {
 
         if(minallowed !== undefined && rng[0] < bounds[0]) range[axrev ? 1 : 0] = minallowed;
         if(maxallowed !== undefined && rng[1] > bounds[1]) range[axrev ? 0 : 1] = maxallowed;
+
+        if(range[0] === range[1]) {
+            if(minallowed !== undefined) range[axrev ? 1 : 0] = ax.l2r(minallowed + 1);
```

Comment:
What if maxallowed-minallowed<1?

Reply:
Good call.
Addressed in https://github.com/plotly/plotly.js/pull/6796/commits/aadbd545563cf16e8e786c90e5a97fa12cef91b7